### PR TITLE
doc: Update the release process

### PR DESCRIPTION
**Pull request by stevenhorsman**
- Reflect the need to update the versions in the Helm Chart
- Add the lock branch instruction
- Add clarity about the permissions needed to complete tasks

### CODE REVIEW

**Review by fidencio — APPROVED** [review]
lgtm, thanks @stevenhorsman!

**Review by GabyCT — APPROVED** [review]

File: docs/Release-Process.md [hunk 1]
```diff
@@ -28,10 +28,20 @@ Bug fixes are released as part of `MINOR` or `MAJOR` releases only. `PATCH` is a
...
+`version` and `appVersion` in the
+[`Chart.yaml`](./../tools/packaging/kata-deploy/helm-chart/kata-deploy/Chart.yaml) file and
+have it merged.
+
+### Lock the `main` branch
+
+In order to prevent any PRs accidentally getting merged during the release process, we
+have recently trailed setting the `main` branch to read only whilst the release action runs.
+
+> **Note:** Admin permission is needed to complete this task.
```
Comment (sprt) [inline docs/Release-Process.md:44]: Nit: maybe we can use Github Markdown for these notes?

https://github.com/orgs/community/discussions/16925

File: docs/Release-Process.md [hunk 2]
```diff
@@ -28,10 +28,20 @@ Bug fixes are released as part of `MINOR` or `MAJOR` releases only. `PATCH` is a
...
 When the `kata-containers/kata-containers` repository is ready for a new release,
-first create a PR to set the release in the `VERSION` file and have it merged.
+first create a PR to set the release in the [`VERSION`](./../VERSION) file and update the
+`version` and `appVersion` in the
+[`Chart.yaml`](./../tools/packaging/kata-deploy/helm-chart/kata-deploy/Chart.yaml) file and
+have it merged.
+
+### Lock the `main` branch
+
+In order to prevent any PRs accidentally getting merged during the release process, we
```
Comment (sprt) [inline docs/Release-Process.md:41]: Should we explain why PRs getting merged during release is a problem?

File: docs/Release-Process.md (hunk 1 above)
Comment (stevenhorsman) [reply docs/Release-Process.md:44]: Nice - I didn't realise there was an updated formatting. I will adopt that now. Thanks for the pointer

File: docs/Release-Process.md (hunk 2 above)
Comment (stevenhorsman) [reply docs/Release-Process.md]: In theory I'm not against PRs going it as if they pass the tests they should be ready to release, but it does interrupt the caches, so I mentioned the release slowing.

**Review by amshinde — APPROVED** [review]

**Review by sprt — APPROVED** [review]

**Pull request merged**